src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -246,7 +250,17 @@ private boolean processLogLine(AtomicInteger totalBytesRead,
                     key -> CloudWatchAttemptLogInformation.builder()
                             .componentName(componentName)
                             .build());
-            reachedMaxSize = addNewLogEvent(totalBytesRead, attemptLogInformation, data.toString());
+            Matcher matcher = textTimestampPattern.matcher(data);
+            Instant logTimestamp = Instant.now();
+            if (matcher.find()) {
+                try {
+                    logTimestamp = Instant.parse(matcher.group());
+                } catch (DateTimeParseException e) {
+                    logger.atTrace().cause(e).log("Unable to parse timestamp for log line");
+                }
+            }
+            reachedMaxSize = addNewLogEvent(totalBytesRead, attemptLogInformation, data.toString(),
+                    logTimestamp.toEpochMilli());
```

Comment:
Nothing is logged if no timestamp is found. Is this expected? If we do find one but parsing fails, is trace really appropriate?

Reply:
I think not logging is fine. It is not that important. If the parsing fails I kept it at trace since the log fill will be full of exceptions since it will log for all log lines. It is too noisy. Also, I feel it is not that important to log this error at error level.